### ISSUE TITLE
Runtime error: Uncaught SyntaxError: import declarations may only appear at top level of a module

### ISSUE THREAD

**Issue by NullVoxPopuli** [ember-template-imports/ember-template-imports#11]
Compiled output: 
![image](https://user-images.githubusercontent.com/199018/111043342-0d27fe00-8410-11eb-8684-189036ffd7cd.png)

looks unchanged from input
```jsx
import { tracked } from '@glimmer/tracking';
import { action } from '@ember/object';
import { hbs } from 'ember-template-imports';
import { helper } from '@ember/component/helper';

export class Switch {
  @tracked on = Boolean(Math.round(Math.random()));

  @action
  flip() {
    this.on = !this.on;
  }
}

const inputId = helper(() => (Math.random() * 0xfffff * 1000000).toString(16).split('.')[0]);

export default hbs`
  {{#let (inputId) as |id|}}
    <input
      {{on 'change' @onClick}}
      class="light-switch"
      type="checkbox"
      checked={{@on}}
      id={{id}}
    >
    <label for={{id}}>
      <div class="screw"></div>
      <div class="switch"></div>
      <div class="screw"></div>
    </label>
  {{/let}}
`;
```

**Comment by NullVoxPopuli** [comment]
if I switch to `export default <template>`,
I end up with a kinda half-transform?

![image](https://user-images.githubusercontent.com/199018/111044295-9988ef80-8415-11eb-940a-41a43b1cf0ff.png)

**Comment by ef4** [comment]
I'm seeing this too in a newly-generate ember 3.25.1 app. In my case I tried to use `<template>` syntax in a gjs file for tests, and I'm ending up with a partially-compiled result in the test.js bundle:

```js
// tests/integration/components/foo-test.js
import { module, test } from 'qunit';
import { setupRenderingTest } from 'ember-qunit';
import { render } from '@ember/test-helpers';
import Component from '@glimmer/component';

module('Integration | Component | foo', function (hooks) {
  setupRenderingTest(hooks);

  test('it renders', async function (assert) {
    class Foo extends Component {
      get message() {
        return 'it works';
      }
      <template>the message is {{this.message}}</template>
    }
    await render(<template><Foo/></template>);
    assert.equal(this.element.textContent.trim(), 'template block text');
  });
});
```

<img width="765" alt="Screen Shot 2021-03-17 at 5 51 53 PM" src="https://user-images.githubusercontent.com/319282/111543485-7fb71780-8749-11eb-893c-bbd6a64329fc.png">

**Comment by pzuraq** [comment]
hmm, ok so the `gjs` plugin is working in your case @ef4, but it's not getting the transformation handled by `ember-cli-htmlbars`. I'll have to dig into that a bit, that's odd.

**Comment by ef4** [comment]
```
[~/hacking/sample]$ npm explain ember-cli-htmlbars                                                                           *[master][ruby-2.7.1]
ember-cli-htmlbars@5.6.5 dev
node_modules/ember-cli-htmlbars
  dev ember-cli-htmlbars@"^5.3.2" from the root project
  ember-cli-htmlbars@"^5.2.0" from @ember/test-helpers@2.2.3
  node_modules/@ember/test-helpers
    dev @ember/test-helpers@"^2.2.0" from the root project
    peer @ember/test-helpers@"^2.1.0" from ember-qunit@5.1.4
    node_modules/ember-qunit
      dev ember-qunit@"^5.1.2" from the root project

ember-cli-htmlbars@3.1.0 dev
node_modules/ember-welcome-page/node_modules/ember-cli-htmlbars
  ember-cli-htmlbars@"^3.0.1" from ember-welcome-page@4.0.0
  node_modules/ember-welcome-page
    dev ember-welcome-page@"^4.0.0" from the root project
```

**Comment by NullVoxPopuli** [comment]
Here is a tiny repro if that helps: https://github.com/NullVoxPopuli/ember-statechart-component/pull/13/files#
Tried to add a gjs demo to my addon's dummy app

**Comment by NullVoxPopuli** [comment]
This is resolved with v2.0.0+
